### ISSUE TITLE
Allow multiple function plot in single subplot

### ISSUE THREAD

**Issue by nisargshah2000** [rajp152k/Compilers-II#14]
Currently we have following code in basic3.fu,
y <- [x^2, sin(x), e^x] here, x^2 go to first subplot, sin(x) go to second and so on.
subplot 2, 1 {
        w <- [k^10, sec(k)^2]
    }
Here, there are two function go to same subplot.
We also need to support the same when user does not write separate subplot.
for example, y <- [[x^2,x+2],x^3] 
x^2 and x+2 both go to subplot(0,0).

**Comment by vedantsingh1404** [comment]
What level of nesting are we planning to support?
If the user gives 
```
y <- [[[x^2 + sin(x)], x^3], 3x + 10]
```
What do we plan to do with this?
```
y <- [x^2, sin(x), e^x] 
```
In this, the plots go to different subplots if the size is specified. If not, they all go to the same plot.

``` 
y <- [[x^2, x + 2], x^3]
```
In this, why should x^2 and x + 2 go to the same plot. Maybe the user wants to split them into smaller subplots.

@rajp152k Thoughts?

**Comment by nisargshah2000** [comment]
What I thought is the following:
let size is (2,1)
Case 1. `y<-[x^2,x+2]`
               x^2 and x+2 go to different subplots.
case 2: `y<-[[x^2,x+2],x^3]`
              x^2 and x+2 go to same subplot and x^3 go to another.
So, basically I thinking one level of nesting in the list for making more than one graph in single subplot.
@vedantsingh1404 `[[[x^2 + sin(x)], x^3], 3x + 10]`  has no meaning.

**Comment by vedantsingh1404** [comment]
Are you also planning on providing support for an array of ranges?

**Comment by nisargshah2000** [comment]
No.

**Comment by vedantsingh1404** [comment]
Yeah maybe it is a good idea to add this. But if we do add this, we might as well add support for multiple ranges. 
```
y <- [x^2, sin(x)]
x <- [10 in 1 to 2, 50 in -1 to 1]
```

Wouldn't this be more useful?

**Comment by nisargshah2000** [comment]
It is better to have common range of x. Only function y is changing. If user wants, he can create separate subplot. 10 in 1 to 2 consumes lots of space and decrease readability.

**Comment by karan121bhukar** [comment]
i think this functionality is good, one might want to see different plots on different scales.

**Comment by rajp152k** [comment]
1-deep nesting seems to be good idea but think we shouldn't enforce a single scale.
can make the defaults that way but different scales should be supported

**Comment by PanchalShubham** [comment]
Aren't we supporting piece wise functions in terms of single variable?
e.g. f(x) = {x; if x is even OR x*2; if x is odd }
something like this?
